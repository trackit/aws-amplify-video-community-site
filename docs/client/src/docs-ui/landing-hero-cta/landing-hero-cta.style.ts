import {css} from "emotion";
import {MQFablet} from "../../amplify-ui/styles/media";

export const platformsGroupStyle = css`
  margin-top: 1.5rem;
  display: flex;
  flex-direction: row;
  justify-content: center;

  stencil-route-link > a {
    display: block;
    padding: 0 0.25rem;
  }

  ${MQFablet} {
    stencil-route-link > a {
      padding: 0 1.25rem;
    }
  }
`;

export const platformIcon = css`
  width: 2.5rem;
  height: 2.5rem;
`;

export const buttonStyle = css`
  margin-top: 2.5rem;

  a {
    margin: 1.5rem auto 0 auto;
  }
`;
