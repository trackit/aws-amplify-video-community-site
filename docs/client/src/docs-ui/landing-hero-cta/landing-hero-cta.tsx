import {Component, h, Host} from "@stencil/core";
import {
  platformsGroupStyle,
  platformIcon,
  buttonStyle,
} from "./landing-hero-cta.style";

@Component({tag: "docs-landing-hero-cta", shadow: false})
export class DocsLandingHeroCTA {
  render() {
    return (
      <Host>
        <docs-internal-link-button
          class={buttonStyle}
          href="/start"
          QSPs="?sc_icampaign=start&sc_ichannel=docs-home"
        >
          <span slot="text">Get started for free</span>
        </docs-internal-link-button>
        <div class={platformsGroupStyle}>
          <docs-internal-link
            href="/start/q/integration/react"
            QSPs="?sc_icampaign=react-start&sc_ichannel=docs-home"
            class="scale-up-on-hover"
          >
            <img
              class={platformIcon}
              src="/assets/integrations/react.svg"
              alt="React Icon"
            />
          </docs-internal-link>
          <docs-internal-link
            href="/start/q/integration/vue"
            QSPs="?sc_icampaign=vue-start&sc_ichannel=docs-home"
            class="scale-up-on-hover"
          >
            <img
              class={platformIcon}
              src="/assets/integrations/vue.svg"
              alt="Vue Icon"
            />
          </docs-internal-link>
          <docs-internal-link
            href="/start"
            QSPs="?sc_icampaign=js-start&sc_ichannel=docs-home"
            class="scale-up-on-hover"
          >
            <img
              class={platformIcon}
              src="/assets/integrations/js.svg"
              alt="JS Icon"
            />
          </docs-internal-link>
          <docs-internal-link
            href="/start/q/integration/ios"
            QSPs="?sc_icampaign=ios-start&sc_ichannel=docs-home"
            class="scale-up-on-hover"
          >
            <img
              class={platformIcon}
              src="/assets/integrations/ios.svg"
              alt="iOS Icon"
            />
          </docs-internal-link>
          <docs-internal-link
            href="/start/q/integration/android"
            QSPs="?sc_icampaign=android-start&sc_ichannel=docs-home"
            class="scale-up-on-hover"
          >
            <img
              class={platformIcon}
              src="/assets/integrations/android.svg"
              alt="Android Icon"
            />
          </docs-internal-link>
          <docs-internal-link
            href="/start/q/integration/flutter"
            QSPs="?sc_icampaign=flutter-start&sc_ichannel=docs-home"
            class="scale-up-on-hover"
          >
            <img
              class={platformIcon}
              src="/assets/integrations/flutter.svg"
              alt="Flutter Icon"
            />
          </docs-internal-link>
        </div>
      </Host>
    );
  }
}
