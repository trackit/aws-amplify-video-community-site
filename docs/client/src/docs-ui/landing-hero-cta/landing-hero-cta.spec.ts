import {DocsLandingHeroCTA} from "./landing-hero-cta";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-landing-hero-cta", () => {
  describe("Render logic", () => {
    it("should render", async () => {
      Error.stackTraceLimit = 50;

      expect(
        (
          await newSpecPage({
            components: [DocsLandingHeroCTA],
            html: `<docs-landing-hero-cta />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
