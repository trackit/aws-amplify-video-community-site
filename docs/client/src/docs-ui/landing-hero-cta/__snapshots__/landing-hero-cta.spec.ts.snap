// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-landing-hero-cta Render logic should render 1`] = `
<docs-landing-hero-cta>
  <docs-internal-link-button class="css-8k8r6" href="/start" qsps="?sc_icampaign=start&amp;sc_ichannel=docs-home">
    <span slot="text">
      Get started for free
    </span>
  </docs-internal-link-button>
  <div class="css-1nsan66">
    <docs-internal-link class="scale-up-on-hover" href="/start/q/integration/react" qsps="?sc_icampaign=react-start&amp;sc_ichannel=docs-home">
      <img alt="React Icon" class="css-16ygk4v" src="/assets/integrations/react.svg">
    </docs-internal-link>
    <docs-internal-link class="scale-up-on-hover" href="/start/q/integration/vue" qsps="?sc_icampaign=vue-start&amp;sc_ichannel=docs-home">
      <img alt="Vue Icon" class="css-16ygk4v" src="/assets/integrations/vue.svg">
    </docs-internal-link>
    <docs-internal-link class="scale-up-on-hover" href="/start" qsps="?sc_icampaign=js-start&amp;sc_ichannel=docs-home">
      <img alt="JS Icon" class="css-16ygk4v" src="/assets/integrations/js.svg">
    </docs-internal-link>
    <docs-internal-link class="scale-up-on-hover" href="/start/q/integration/ios" qsps="?sc_icampaign=ios-start&amp;sc_ichannel=docs-home">
      <img alt="iOS Icon" class="css-16ygk4v" src="/assets/integrations/ios.svg">
    </docs-internal-link>
    <docs-internal-link class="scale-up-on-hover" href="/start/q/integration/android" qsps="?sc_icampaign=android-start&amp;sc_ichannel=docs-home">
      <img alt="Android Icon" class="css-16ygk4v" src="/assets/integrations/android.svg">
    </docs-internal-link>
    <docs-internal-link class="scale-up-on-hover" href="/start/q/integration/flutter" qsps="?sc_icampaign=flutter-start&amp;sc_ichannel=docs-home">
      <img alt="Flutter Icon" class="css-16ygk4v" src="/assets/integrations/flutter.svg">
    </docs-internal-link>
  </div>
</docs-landing-hero-cta>
`;
